File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/entity/station_response/Station.java
package com.zuk0.gaijinsmash.riderz.data.local.entity.station_response;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import com.zuk0.gaijinsmash.riderz.data.local.entity.etd_response.Etd;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;

import java.util.List;

@Entity(tableName = "stations")
public class Station {

    @PrimaryKey(autoGenerate = true)
    private int id;

    @ColumnInfo(name = "name")
    @Element
    private String name;

    @ColumnInfo(name = "abbr")
    @Element(required = false)
    private String abbr;

    @ColumnInfo(name = "latitude")
    @Element(name = "gtfs_latitude", required = false)
    private double latitude;

    @ColumnInfo(name = "longitude")
    @Element(name = "gtfs_longitude", required = false)
    private double longitude;

    @ColumnInfo(name = "address")
    @Element(name = "address", required = false)
    private String address;

    @ColumnInfo(name = "city")
    @Element(required = false)
    private String city;

    @ColumnInfo(name = "county")
    @Element(required = false)
    private String county;

    @ColumnInfo(name = "state")
    @Element(required = false)
    private String state;

    @ColumnInfo(name = "zipcode")
    @Element(required = false)
    private String zipcode;

    @ColumnInfo(name = "platform_info")
    @Element(name = "platform_info", required = false)
    private String platformInfo;

    @ColumnInfo(name = "intro")
    @Element(data = true, required = false)
    private String intro;

    @ColumnInfo(name = "cross_street")
    @Element(data = true, name="cross_street", required = false)
    private String crossStreet;

    @ColumnInfo(name = "food")
    @Element(data = true, required = false)
    private String food;

    @ColumnInfo(name = "shopping")
    @Element(data = true, required = false)
    private String shopping;

    @ColumnInfo(name = "attraction")
    @Element(data = true, required = false)
    private String attraction;

    @ColumnInfo(name = "link")
    @Element(data = true, required = false)
    private String link;

    @Ignore
    @ElementList(name="etd", inline=true, required = false)
    private List<Etd> etdList;

    @Ignore
    @ElementList(name = "north_routes", required = false)
    private List<Route> northRoutes;

    @Ignore
    @ElementList(name = "south_routes", required = false)
    private List<Route> southRoutes;

    @Ignore
    @ElementList(name = "north_platforms", required = false)
    private List<Platform> northPlatforms;

    @Ignore
    @ElementList(name = "south_platforms", required = false)
    private List<Platform> southPlatforms;

    public Station () {}

    // Setters
    public void setId(int id) {
        this.id = id;
    }

    public void setName(@NonNull String name) { this.name = name; }
    public void setAddress(String address) { this.address = address; }
    public void setLatitude(double latitude) { this.latitude = latitude; }
    public void setLongitude(double longitude) { this.longitude = longitude; }
    public void setZipcode(String zipcode) { this.zipcode = zipcode; }
    public void setCity(String city) { this.city = city; }
    public void setCounty(String county) { this.county = county; }
    public void setState(String state) { this.state = state; }
    public void setPlatformInfo(String platformInfo) { this.platformInfo = platformInfo; }
    public void setIntro(String intro) { this.intro = intro; }
    public void setCrossStreet(String crossStreet) { this.crossStreet = crossStreet; }
    public void setFood(String food) { this.food = food; }
    public void setShopping(String shopping) { this.shopping = shopping; }
    public void setAttraction(String attraction) { this.attraction = attraction; }
    public void setLink(String link) { this.link = link; }
    public void setAbbr(String abbr) { this.abbr = abbr; }
    public void setEtdList(List<Etd> etdList) {
        this.etdList = etdList;
    }
    public void setNorthRoutes(List<Route> northRoutes) {
        this.northRoutes = northRoutes;
    }
    public void setSouthRoutes(List<Route> southRoutes) {
        this.southRoutes = southRoutes;
    }


    // Getters
    @NonNull
    public int getId() {
        return id;
    }
    public String getName() {
        return name;
    }
    public String getAddress() {
        return address;
    }
    public double getLatitude() {
        return latitude;
    }
    public double getLongitude() {
        return longitude;
    }
    public String getZipcode() { return zipcode; }
    public String getCity() { return city; }
    public String getCounty() { return county; }
    public String getState() { return state; }
    public String getAbbr() { return abbr; }
    public String getLink() { return link; }
    public String getAttraction() { return attraction; }
    public String getShopping() { return shopping; }
    public String getFood() { return food; }
    public String getCrossStreet() { return crossStreet; }
    public String getIntro() { return intro; }
    public String getPlatformInfo() { return platformInfo; }
    public List<Etd> getEtdList() {
        return etdList;
    }
    public List<Route> getNorthRoutes() {
        return northRoutes;
    }
    public List<Route> getSouthRoutes() {
        return southRoutes;
    }

}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/entity/station_response/Station.kt
package com.zuk0.gaijinsmash.riderz.data.local.entity.station_response

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.zuk0.gaijinsmash.riderz.data.local.entity.etd_response.Etd
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList

@Entity(tableName = "stations")
class Station {
    // Setters
    // Getters
    @PrimaryKey(autoGenerate = true)
    var id = 0

    @ColumnInfo(name = "name")
    @field:Element(name = "name", required = false)
    var name: String? = null

    @ColumnInfo(name = "abbr")
    @field:Element(required = false)
    var abbr: String? = null

    @ColumnInfo(name = "latitude")
    @field:Element(name = "gtfs_latitude", required = false)
    var latitude = 0.0

    @ColumnInfo(name = "longitude")
    @field:Element(name = "gtfs_longitude", required = false)
    var longitude = 0.0

    @ColumnInfo(name = "address")
    @field:Element(name = "address", required = false)
    var address: String? = null

    @ColumnInfo(name = "city")
    @field:Element(required = false)
    var city: String? = null

    @ColumnInfo(name = "state")
    @field:Element(required = false)
    var state: String? = null

    @ColumnInfo(name = "county")
    @field:Element(required = false)
    var county: String? = null

    @ColumnInfo(name = "zipcode")
    @field:Element(required = false)
    var zipcode: String? = null

    @ColumnInfo(name = "platform_info")
    @field:Element(name = "platform_info", required = false)
    var platformInfo: String? = null

    @ColumnInfo(name = "intro")
    @field:Element(data = true, required = false)
    var intro: String? = null

    @ColumnInfo(name = "cross_street")
    @field:Element(data = true, name = "cross_street", required = false)
    var crossStreet: String? = null

    @ColumnInfo(name = "food")
    @field:Element(data = true, required = false)
    var food: String? = null

    @ColumnInfo(name = "shopping")
    @field:Element(data = true, required = false)
    var shopping: String? = null

    @ColumnInfo(name = "attraction")
    @field:Element(data = true, required = false)
    var attraction: String? = null

    @ColumnInfo(name = "link")
    @field:Element(data = true, required = false)
    var link: String? = null

    @Ignore
    @field:ElementList(name = "etd", inline = true, required = false)
    var etdList: MutableList<Etd>? = null

    @Ignore
    @field:ElementList(name = "north_routes", required = false)
    var northRoutes: MutableList<Route>? = null

    @Ignore
    @field:ElementList(name = "south_routes", required = false)
    var southRoutes: MutableList<Route>? = null

    @Ignore
    @field:ElementList(name = "north_platforms", required = false)
    var northPlatforms: MutableList<Platform>? = null

    @Ignore
    @field:ElementList(name = "south_platforms", required = false)
    var southPlatforms: MutableList<Platform>? = null

    @ColumnInfo
    @field:Element(name = "message", required = false)
    var message: String? = null

    @ColumnInfo
    @field:Element(name = "error", required = false)
    var error: String? = null

    override fun toString(): String {
        return name ?: ""
    }
}